joomla/joomla-cms
# [5.2] Jooa11y plugin and page cache conflicts

Pull Request for Issue #37068.

### Summary of Changes

1. Jooa11y plugin has issues with IDE (unknown classes and methods):
1.1. `getSubscribedEvents()` throws unknown `\Joomla\Plugin\System\Jooa11y\Extension\Exception`
1.2. `initJooa11y()` should return void
1.3. Force `$document` class for IDE, now `$document->addScriptOptions()` and `$document->getWebAssetManager()` are detected by IDE
2. Extra useless cache files are created if system page cache is enabled.
3. The article updates are not visible on the next accessibility check after article save.

### Testing Instructions

1. Enable Joomla system page cache plugin (browser caching should be disabled in the plugin settings).
2. Ensure that you are not logged in frontend.
6. Clear Joomla cache
7. Load article in frontend
8. See a single `.php` file in `/administrator/cache/page`
9. Edit the same article in backend, click "Accessibility Check" button to explore the checks
10. See another `.php` file in `/administrator/cache/page`
11. Now change the article text, click Save button
12. Click "Accessibility Check" again and see the old article version, no changes are applied.

### Actual result BEFORE applying this Pull Request

System page cache plugin creates an extra file for each accessibility check.
No way to see the changes after article save without clearing 'page' cache manually.

### Expected result AFTER applying this Pull Request

System page cache is not active on accessibility check,
The changes in articles are displayed after article save.

### Link to documentations
Please select:
